shm/inc/Map.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
```

Comment:
Newline. Compiler directive (pragma) s, user header files ("") and standard headers (<>) should be separated from each other. Any other opinions? Why?